url/nodeurl.go

```diff
@@ -0,0 +1,117 @@
+package url
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+type searchParam struct {
+	name  string
+	value string
+}
+
+func (sp *searchParam) Encode() string {
+	return sp.string(true)
+}
+
+func (sp *searchParam) string(encode bool) string {
+	vals := []string{}
+	if encode {
+		vals = append(vals, fmt.Sprintf("%s=%s", url.QueryEscape(sp.name), url.QueryEscape(sp.value)))
+	} else {
+		vals = append(vals, fmt.Sprintf("%s=%s", sp.name, sp.value))
+	}
+	return strings.Join(vals, "&")
```

Comment:
Why do you need the slice and the `Join()` here?

Reply:
We don't need it at all, this was a relic from a previous implementation. I'll simplify this.